jobscheduler/scheduler-component-shared/src/java/org/sakaiproject/component/app/scheduler/jobs/LBCSSFolderHideJob.java

```diff
@@ -0,0 +1,84 @@
+import org.apache.commons.logging.LogFactory;
+import org.quartz.DisallowConcurrentExecution;
+import org.quartz.Job;
+import org.quartz.JobExecutionContext;
+import org.quartz.JobExecutionException;
+import org.sakaiproject.content.api.ContentCollectionEdit;
+import org.sakaiproject.content.api.ContentHostingService;
+import org.sakaiproject.db.api.SqlService;
+import org.sakaiproject.entity.api.ResourceProperties;
+import org.sakaiproject.entity.api.ResourcePropertiesEdit;
+import org.sakaiproject.event.api.UsageSessionService;
+import org.sakaiproject.exception.IdUnusedException;
+import org.sakaiproject.exception.InUseException;
+import org.sakaiproject.exception.PermissionException;
+import org.sakaiproject.exception.TypeException;
+import org.sakaiproject.tool.api.Session;
+import org.sakaiproject.tool.api.SessionManager;
+
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.util.List;
+
+/**
+ * Job to make LB-CSS folder hidden but contents accessible.
+ * LB-CSS folder should not be visible to access user, this job will hide all existing non-hidden LB-CSS folder.
+ * @author neelam 
+ *
+ */
+@DisallowConcurrentExecution
+public class LBCSSFolderHideJob implements Job {
+
+	private SqlService sqlService;
+	private SessionManager sessionManager;
+
+	private static final Log LOG = LogFactory.getLog(LBCSSFolderHideJob.class);
+	
+	private ContentHostingService contentHostingService;
+
+	public void setSqlService(SqlService sqlService) {
+		this.sqlService = sqlService;
+	}
+
+	public void setSessionManager(SessionManager sessionManager) {
+		this.sessionManager = sessionManager;
+	}
+
+	public void setContentHostingService(ContentHostingService contentHostingService) {
+		this.contentHostingService = contentHostingService;
+	}
+
+	public void execute(JobExecutionContext context) throws JobExecutionException {
+		//set current user in session
+		Session session = sessionManager.getCurrentSession();
+		session.setUserEid("admin");
+		session.setUserId("admin");
+		//sql to get collection ids for all LB-CSS folder
+		String sql = "SELECT COLLECTION_ID FROM CONTENT_COLLECTION WHERE COLLECTION_ID LIKE \"%LB-CSS%\" ";
+		LOG.debug("SQL to get collection Id for LB-CSS folder " + sql);
+		List<String> collection_ids =  sqlService.dbRead(sql);
+		for(String id : collection_ids){
+			try {
+				ContentCollectionEdit contentCollectionEdit = contentHostingService.editCollection(id);
+				ResourcePropertiesEdit resourceProperties = contentCollectionEdit.getPropertiesEdit();
+				//if LB-CSS folder is not hidden, hide it
+				if(!("true".equals(resourceProperties.getProperty(ResourceProperties.PROP_HIDDEN_WITH_ACCESSIBLE_CONTENT)))){
+					resourceProperties.addProperty(ResourceProperties.PROP_HIDDEN_WITH_ACCESSIBLE_CONTENT, "true");
+				}
+				contentHostingService.commitCollection(contentCollectionEdit);
+			} catch (IdUnusedException e) {
+				LOG.error("IdUnusedException " + e);
+			} catch (TypeException e) {
```

Comment:
You can collapse all these Exceptions down by using SakaiException class and if any of them don't extends SakaiException, multicatch.